androidshared/src/main/java/org/odk/collect/androidshared/system/UriExt.kt

```diff
@@ -0,0 +1,38 @@
+package org.odk.collect.androidshared.system
+
+import android.content.ContentResolver
+import android.net.Uri
+import android.provider.OpenableColumns
+import timber.log.Timber
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+fun Uri.toFile(contentResolver: ContentResolver, dest: File) {
+    try {
+        contentResolver.openInputStream(this)?.use { inputStream ->
+            FileOutputStream(dest).use { outputStream ->
+                inputStream.copyTo(outputStream)
+            }
+        }
+    } catch (e: IOException) {
+        Timber.e(e)
+    }
+}
+
+fun Uri.getFileName(contentResolver: ContentResolver): String? {
+    var fileName: String? = null
+    if (scheme == ContentResolver.SCHEME_CONTENT) {
```

Comment:
As far as I can tell, this `if` block isn't tested anywhere. Is it even needed? If so, maybe it's worth creating a unit test given it looks like the other case is the simpler one that gets used in the offline maps tests.

Reply:
Actually, there are 3 types of schema (`SCHEME_CONTENT`, `SCHEME_FILE`, `SCHEME_ANDROID_RESOURCE`). I believe `Intent(Intent.ACTION_GET_CONTENT)`  (the one I've created this method for) should return uris with `SCHEME_CONTENT` only but maybe it also could be `SCHEME_FILE` (it might depend on the source of selected files - local storage, Google Drive, etc).
Since this method should handle all the cases (so that we can use it in other places with other contexts) I've extended it.
I've also added some tests for all the methods in `UriExt` class.
Those tests are not sufficient because to cover all the cases I would need to mock a lot of stuff (just look at the method body) but I will think about it.
If it's the last thing you want to change in this pr let's merge and not block it. I will try to add more tests in the third pr or file a separate issue.